be/src/util/es_search_builder.cpp

```diff
@@ -0,0 +1,63 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "es_search_builder.h"
+#include<sstream>
+#include <boost/algorithm/string/join.hpp>
+#include "common/logging.h"
+#include "rapidjson/document.h"
+#include "rapidjson/stringbuffer.h"
+#include "rapidjson/writer.h"
+
+namespace doris {
+
+SearchRequestBuilder::SearchRequestBuilder() {
+
+}
+
+SearchRequestBuilder::~SearchRequestBuilder() {
+    
+}
+
+std::string SearchRequestBuilder::build() {
+    rapidjson::Document es_query_dsl;
+    rapidjson::Document::AllocatorType &allocator = es_query_dsl.GetAllocator();
+    es_query_dsl.SetObject();
+    if (_fields.size() > 0) {
+        rapidjson::Value source_node(rapidjson::kArrayType);
+        for (auto iter = _fields.cbegin(); iter != _fields.cend(); iter++) {
```

Comment:
why do you call `cbegin` other `begin`. Could you add some comment for this?

Reply:
I will replace cxxx with xxx